SeafarersAPI/Data/Repositories/SubmissionRepository.cs

```diff
@@ -3,56 +3,57 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
 
 namespace SeafarersAPI.Data.Repositories
 {
-    public class SubmissionRepository
+    public class SubmissionRepository : IRepository<Submission>
     {
         public SubmissionContext context;
         public SubmissionRepository()
         {
-            if(context== null)
+            if(context == null)
             {
                 context = new SubmissionContext();
             }
         }
 
-        public List<Submission> GetSubmissions()
+        public IEnumerable<Submission> GetList(int startIndex, int amountToGet)
         {
-            return context.Submissions.ToList();
+            return context.Submissions.Skip(startIndex).Take(amountToGet).ToList();
         }
 
-        public Submission GetSubmissionById(int submissionId)
+        public int Save(Submission submission)
         {
-            var submission = context.Submissions.Where(x => x.SubmissionId == submissionId).SingleOrDefault();
-
-            return submission;
+            // TODO: Add locking mechanism or we'll run into concurrency issues
+            context.Submissions.Add(submission);
+            return context.SaveChanges();
         }
 
-        public List<Submission> GetSubmissionsByCDNnumber(string cdn_number)
+        public int Delete(Submission submission)
         {
-            return context.Submissions.Where(x => x.CdnNumber == cdn_number).ToList();
+            // TODO: Add locking mechanism or we'll run into concurrency issues
+            context.Entry(submission).State = EntityState.Deleted;
+            return context.SaveChanges();
```

Comment:
We want to return the results of the save here so that we can handle errors in the case nothing was saved. We also want to add a locking mechanism to avoid concurrency problems in the future.